tfhe/src/core_crypto/algorithms/lwe_multi_bit_programmable_bootstrapping.rs

```diff
@@ -1680,11 +1606,25 @@ pub fn std_multi_bit_deterministic_blind_rotate_assign<Scalar, InputCont, Output
             );
 
             let dest_idx = thread_id;
-            for (lwe_mask_elements, ggsw_group) in
-                work_queue.iter().skip(thread_id).step_by(thread_count.0)
-            {
-                let (ready_for_consumer_lock, condvar, fourier_ggsw_buffer) =
-                    &fourier_multi_bit_ggsw_buffers[dest_idx];
+
+            #[allow(clippy::type_complexity)]
+            let (ready_for_consumer_lock, condvar, fourier_ggsw_buffer): &(
+                Mutex<bool>,
+                Condvar,
+                Mutex<FourierGgswCiphertext<ABox<[c64]>>>,
+            ) = &fourier_multi_bit_ggsw_buffers[dest_idx];
+
+            for i in 0.. {
+                let work_index = thread_id + i * thread_count.0;
+                if work_index >= max_work_index {
+                    break;
+                }
```

Comment:
prefer the construction without an infinite range and proper stepping, because we know when the loop is supposed to end

Reply:
A while loop would be clearer IMO
Will change

Reply:
Done